packages/axelar-soroban-std/src/upgrade.rs

```diff
@@ -1,10 +1,77 @@
-use soroban_sdk::{contractclient, BytesN, Env, String};
+use crate::ensure;
+use soroban_sdk::{contractclient, contracttype, Address, BytesN, Env, String, Symbol};
 
 #[contractclient(name = "UpgradeableClient")]
 pub trait UpgradeableInterface {
     /// Returns the current version of the contract.
     fn version(env: &Env) -> String;
 
+    /// The owner that is allowed to upgrade the contract.
+    fn owner(env: &Env) -> Address {
+        env.storage()
+            .instance()
+            .get(&DataKey::Owner)
+            .expect("owner not found")
+    }
+
```

Comment:
add a default trait fn `set_owner` to set the owner. The contract constructors can then use `Self::set_owner` to set the owner instead of accessing the `upgrade::DataKey` directly. Default trait functions aren't included in contractimpl entrypoints last time I checked. It works here, although `owner` method above can't be called to query the owner on-chain